### PR TITLE
Search and replace pagination

### PR DESCRIPTION
**Pull request by t-zilla**
## Changes in this pull request  
Results grid in File -> Search and Replace Assignments now supports pagination.

### CODE REVIEW

**Comment by kubaplas** [comment]
@t-zilla Getting a list of objects containing relation should be rewritten in a way that would limit the amount of objects using db query. Now it still collects the data about all of the relations, still resulting in communication failure.

**Comment by dvesh3** [comment]
@t-zilla could you please address this concern? https://github.com/pimcore/pimcore/pull/6207#issuecomment-614097816

File: models/Dependency/Dao.php [hunk 1]
```diff
@@ -196,6 +197,64 @@ public function getRequiredBy($offset = null, $limit = null)
...
+        if (!in_array($orderBy, ['id', 'type', 'path'])) {
+            $orderBy = 'id';
+        }
+
+        if (!in_array($orderDirection, ['ASC', 'DESC'])) {
+            $orderDirection = 'ASC';
+        }
+
+        $query = "
+            SELECT id, type, path
```
Comment (dvesh3) [inline models/Dependency/Dao.php:226]: @t-zilla can you please explain what is the benefit of using such a complex query over a simple one in `getRequiredBy()`? AFAIK we still need to fetch the relation elements to check permissions and so.. or i missed a point here?

Comment (t-zilla) [reply models/Dependency/Dao.php:226]: To enable pagination without a need to fetch _all_ dependencies which causes a communication error down the line. The simple query on `dependencies` table doesn't enable us to sort by `path` which appears on the S&R grid.

Comment (dvesh3) [reply models/Dependency/Dao.php:226]: Thanks for your explanation! I believe we can keep it then as it is just used in S&R grid.

File: models/Dependency/Dao.php [hunk 2]
```diff
@@ -196,6 +197,64 @@ public function getRequiredBy($offset = null, $limit = null)
...
     }
 
+    /**
+     * @param null $orderBy
+     * @param null $orderDirection
+     * @param null $offset
+     * @param null $limit
+     * @return array
+     */
+    public function getRequiredByWithPath($orderBy = null, $orderDirection = null, $offset = null, $limit = null)
```
Comment (dvesh3) [inline models/Dependency/Dao.php:207]: I would still keep offset and limit params in first place to be consistent with `getRequiredBy()`

Comment (t-zilla) [reply models/Dependency/Dao.php]: Done

**Comment by dvesh3** [comment]
@t-zilla please revert all translation files except en.json. Translations (except English) are managed in POEditor: https://pimcore.com/en/resources/translations

File: models/Dependency/Dao.php [hunk 3]
```diff
@@ -196,6 +197,64 @@ public function getRequiredBy($offset = null, $limit = null)
         return $requiredBy;
     }
 
+    /**
+     * @param null $orderBy
```
Comment (dvesh3) [inline models/Dependency/Dao.php:201]: always null?

**Comment by dvesh3** [comment]
@t-zilla nice work! thanks a lot 👍

**Pull request merged**